### ISSUE TITLE
Visitor-facing answer agent: gated on an ADR 0004 decision

### ISSUE THREAD

**Issue by adamgreenwell** [adamgreenwell/wayfindr#764]
Nothing answers a visitor before a human does. Competitors deflect a substantial share of volume this way — Intercom's Fin, Chatwoot's Captain, Tidio's Lyro, Crisp's AI — and it is exactly the volume a small self-hosted team is least able to staff.

**This issue is a decision, not a task.** It should not be picked up as build work in its current state.

## Why it is gated

ADR 0004 states that *"autonomous support replies, broad customer-data analysis, and background training/export workflows are out of scope until the privacy, audit, consent, and evaluation model is much more mature."*

A visitor-facing answer agent **is** an autonomous support reply. Building one means amending or superseding ADR 0004 first. That is a legitimate thing to do — ADRs record decisions at a point in time and the competitive position has moved since May 2026 — but it is a deliberate act, and doing it implicitly by shipping the feature would be the wrong way round.

## The precondition that actually blocks it

Of the four things ADR 0004 requires to mature, three already have substantial machinery:

- **Privacy** — `docs/privacy/`, the data inventory, the cobrowse boundary.
- **Audit** — `AuditEvent`, account-visible export, the break-glass trail.
- **Consent** — the cobrowse request/grant lifecycle, the most carefully built consent model in the product.

**Evaluation has nothing.** No harness, no fixtures, no accuracy measure, no regression signal for generated output anywhere in the codebase.

That gap is not incidental. An autonomous agent without evaluation is a system that can get worse without anyone noticing, in front of customers, on someone else's install — and a self-hosted product cannot watch a dashboard and roll back centrally when it does. **Evaluation is the load-bearing precondition**, and it is the first item below for that reason.

## Proposed scope, in order

The first two produce the evidence the third needs. Nothing after the ADR decision should start before it.

- [x] **An evaluation harness** — a fixture set of real support questions with known-good answers, an accuracy and refusal measure, and a regression signal that runs without live provider keys. This is useful on its own even if the answer agent is never built, because #763 needs it too.
- [x] **A confidence and refusal model** — the answer agent's most important behavior is declining to answer. Measure that before anything else, because a system that guesses confidently is worse than no system.
- [x] **An ADR decision** — amend, supersede, or reaffirm ADR 0004 with the evaluation evidence in hand. If the honest answer is that the deflection is not good enough to put in front of customers, reaffirming the deferral is a real and acceptable outcome.

Only if that decision permits it:

- [ ] **Answers grounded strictly in published #743 articles**, with the source cited in the reply. An agent that can only repeat what the operator wrote is a much smaller trust problem than one that reasons freely over transcripts.
- [ ] **Handoff on low confidence**, with full context, defaulting to a human. The handoff is the feature; the answering is the optimization.
- [ ] **Per-site enable, off by default**, and visible to the visitor that they are talking to an automated agent.
- [ ] **Audit of every autonomous reply**, per ADR 0004's requirement that AI actions affecting support records are auditable.

Out of scope regardless of the decision:

- [ ] Taking actions on a customer's behalf — refunds, account changes, anything with a side effect. Fin does this; it is a substantially larger trust and audit problem and should not ride along.
- [ ] Answering from transcripts rather than published articles. Transcripts contain other customers' data, and grounding generation in them is precisely the "broad customer-data analysis" the ADR defers.
- [ ] Training or fine-tuning on support data. Deferred by ADR 0004 and it should stay deferred.
- [ ] Voice.

## A note on the competitive pressure

The honest read from the competitive analysis is that this became a buyer filter during 2025–26 rather than a differentiator, and there is real pressure to have a checkbox here.

That pressure is not a reason to skip the gate. Wayfindr's position is a support platform for people who care where their data goes, and a poorly-evaluated autonomous agent damages that position far more than its absence does. **"We do not do this yet, and here is why" is a coherent answer to a buyer.** "We do this badly" is not.

Part of #762, #743

**Comment by adamgreenwell** [comment]
The first gate is complete in #922 (merged as `1a2e03a4`), so I checked only **An evaluation harness** above.

Evidence shipped:

- provider-free `wayfindr:ai-evaluate` command with human and JSON reports
- nine realistic, wholly synthetic answer/refusal cases plus curated known-good responses; private local fixture/response files are also supported without committing customer data
- answer accuracy, refusal recall, citation precision/recall, fact coverage, unsafe-answer rate, and threshold exit codes
- strict bounded/versioned inputs, grounded expected facts, privacy-safe failure output, whole-token fact matching, and refusal-payload detection
- focused suite: 8 tests / 66 assertions
- amended local full suite: 3,616 tests / 33,888 assertions
- exact-head hosted CI: all five jobs green, including PostgreSQL
- exact-head Codex review: no major issues after fixing its P1 token-boundary and P3 ID-length findings

Boundary held: this adds no visitor answer runtime, provider call, customer transcript, confidence policy, or ADR change. **A confidence and refusal model** and **An ADR decision** remain intentionally unchecked and are the next two gates.

**Comment by adamgreenwell** [comment]
Confidence/refusal gate update: PR #923 merged to `main` as `33a8ea2c` after all five CI jobs passed and an exact-head Codex re-review found no major issues.

What is now shipped:

- an explicit 80% policy threshold that converts lower-confidence answer candidates into human handoffs
- strict answer/refusal decisions and refusal-reason labels
- measurements for candidate vs policy decisions, answer accuracy/coverage, refusal recall/reason accuracy, citation quality, unsafe answers, overconfident errors, unwarranted handoffs, and Brier calibration
- an opt-in `wayfindr:ai-evaluate:capture` command for complete synthetic provider runs, with stable provider/model metadata, no ground truth in prompts, private exclusive output, and no committed provider responses
- 31 focused tests / 168 assertions after review fixes; the broader local and hosted suites remain green

Evidence boundary: this checkout still reports the provider configuration as `unavailable` with no provider, model, endpoint, or API key, and no Ollama/local model executable is present. I therefore did **not** run a real model, check **A confidence and refusal model**, or change ADR 0004. The machinery to collect that evidence is merged; the evidence itself still requires an explicitly configured provider and this command:

```bash
cd apps/server
php artisan wayfindr:ai-evaluate:capture \
  --allow-provider \
  --output=/absolute/private/path/grounded-answer-responses.json \
  --json
php artisan wayfindr:ai-evaluate \
  --responses=/absolute/private/path/grounded-answer-responses.json \
  --json
```

**Comment by adamgreenwell** [comment]
Stage-provider recheck on 8 September 2026, after access to the owning Forge organization was restored while resolving #814:

- stage is running `33a8ea2c`, which includes the provider-capture and scoring commands from #923
- **Operator → Agent copilot** reports **Optional — not configured**
- provider is **None (keep every workflow manual)**, with no model or custom endpoint configured

I did not alter provider settings, source credentials elsewhere, or make a model call. This means the deployed runtime is ready to collect the confidence/refusal evidence, but it still has no approved model to measure. The confidence/refusal checkbox and ADR decision therefore remain correctly unchecked.

**Comment by adamgreenwell** [comment]
OpenRouter provider evaluation completed on stage on 8 September 2026.

Deployment and routing evidence:

- PR #925 is merged and deployed at exact SHA `613f5ae66a51447b4d9a7bf070d8defc83cef016`; London, New York, and Singapore health checks returned 200.
- Every OpenRouter request was pinned to one upstream provider, with fallbacks disabled and zero-data-retention requested. This follows the documented [provider routing](https://openrouter.ai/docs/guides/routing/provider-selection) and [ZDR](https://openrouter.ai/docs/guides/features/zdr) controls.
- Only the nine wholly synthetic evaluation cases and synthetic connection probes were sent. No transcript, ticket, visitor detail, attachment, or other customer data was used.
- The evaluation key has a $5 lifetime cap and expires 15 September 2026. Current measured usage is $0.073.
- Provider outputs remain private server-side files outside the repository, mode 0600.

Comparison results:

| Model / pinned upstream | Capture | Passed | Key findings |
| --- | --- | ---: | --- |
| `openai/gpt-5.2` / `azure` | complete | 8/9 | 100% answer accuracy and coverage, 100% refusal recall, 100% citation precision/recall, 0% unsafe and overconfident answers; one medical-refusal reason mismatch (75% refusal-reason accuracy) |
| `openai/gpt-5.6-sol` / `azure` | complete | 7/9 | 80% answer accuracy, 100% refusal recall/reason accuracy, one missing fact, 20% overconfident-error rate, Brier 22 |
| `google/gemini-3.8-flash` / `google-vertex/global` | complete | 6/9 | 40% answer accuracy, 100% refusal recall/reason accuracy, three missing-fact/overconfident errors, Brier 32.31 |
| `anthropic/claude-sonnet-5` / `amazon-bedrock/global` | invalid | n/a | Connection passed, but the structured capture became invalid at `medical-advice-request`; no result file was written |
| `anthropic/claude-sonnet-4.5` / `amazon-bedrock` | invalid | n/a | Connection passed, but the structured capture became invalid at `password-reset-link`; no result file was written |

Stage is left on the strongest observed candidate: `openai/gpt-5.2` through `openrouter/azure`; a fresh synthetic connection probe passed after re-selection.

Conclusion: the confidence/refusal machinery is now both implemented and exercised against real hosted models, so that gate can be considered complete. The evidence does **not** justify an autonomous visitor-facing rollout yet: no candidate cleared the full policy, and the best candidate still missed one refusal-reason requirement. ADR 0004 therefore remains unchanged and its decision gate stays open. The sensible next slice is a narrow review of that refusal-reason mismatch plus one direct-provider finalist run, then an explicit ADR decision—not answer-agent build work.

**Comment by adamgreenwell** [comment]
Post-deployment GPT-5.2 rerun completed on 8 September 2026.

- PR #926 merged as `ae2bee5ab77607d5bf87ac04c87612f8b17180b7` after all five hosted CI lanes passed and exact-head Codex review found no major issues.
- Canonical `main`, the `northcoastmedia/wayfindr` deployment fork, and Forge active release all matched that exact SHA before evaluation.
- `https://wayfindr.on-forge.com/up` returned HTTP 200.
- A fresh private capture completed all nine wholly synthetic cases through `openrouter/azure` using `openai/gpt-5.2`.
- The output file is outside the repository with mode 0600.
- Result: **passed, 9/9 cases**.
- Answer accuracy/coverage, refusal recall/reason accuracy, citation precision/recall, and fact coverage were all 100%.
- Unsafe-answer, overconfident-error, and unwarranted-handoff rates were all 0%.
- Brier score was 0.31 against the maximum of 5.

The fix did not change fixtures, expected outcomes, thresholds, parser rules, or scorer behavior. It defined the already-existing refusal taxonomy and its precedence in the provider prompt. The previous medical case had refused safely at 0% confidence but selected `sensitive_request`; after the contract clarification it selected the required high-risk path and the complete suite passed.

Evidence boundary: this is one current model/provider route over nine synthetic cases. It is strong enough to feed the explicit ADR 0004 decision, but it is not by itself production approval or evidence from real customer conversations. The ADR checkbox remains open until that decision is recorded.

**Comment by adamgreenwell** [comment]
Decision completed by #927, merged as `086438b1` after all five hosted CI lanes passed and exact-head Codex review found no major issues.

ADR 0004 now records the September 8 reassessment and **reaffirms the deferral of autonomous visitor replies**. The agent-controlled copilot remains the approved boundary. The green 9/9 GPT-5.2 synthetic run is retained as evidence, not promoted into production approval.

This closes #764 because the issue was explicitly a decision, not because the deferred answer agent shipped. The conditional implementation items remain undone and out of scope under the reaffirmed ADR.
